src/test/controllerscriptenginelegacy_test.cpp

```diff
@@ -620,3 +620,277 @@ TEST_F(ControllerScriptEngineLegacyTest, connectionExecutesWithCorrectThisObject
     // The counter should have been incremented exactly once.
     EXPECT_DOUBLE_EQ(1.0, pass->get());
 }
+
+TEST_F(ControllerScriptEngineLegacyTest, beginTimer_repeatedTimer) {
+    auto co = std::make_unique<ControlPotmeter>(ConfigKey("[Test]", "co"),
+            -10.0,
+            10.0);
+    co->setParameter(0.0);
+    EXPECT_TRUE(evaluateAndAssert(
+            "engine.setValue('[Test]', 'co', 0.0);"));
+    EXPECT_DOUBLE_EQ(0.0, co->get());
+
+    EXPECT_TRUE(
+            evaluateAndAssert(R"(engine.beginTimer(50, function() {
+                                    let x = engine.getValue('[Test]', 'co');
+                                    x++; 
+                                    engine.setValue('[Test]', 'co', x);
+                                 }, false);)"));
+    processEvents();
+    EXPECT_DOUBLE_EQ(0.0, co->get());
+
+    cEngine->thread()->msleep(70);
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(1.0, co->get());
+
+    cEngine->thread()->msleep(140);
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(2.0, co->get());
+}
+
+TEST_F(ControllerScriptEngineLegacyTest, beginTimer_singleShotTimer) {
+    auto co = std::make_unique<ControlPotmeter>(ConfigKey("[Test]", "co"),
+            -10.0,
+            10.0);
+    co->setParameter(0.0);
+    EXPECT_TRUE(evaluateAndAssert(
+            "engine.setValue('[Test]', 'co', 0.0);"));
+    EXPECT_DOUBLE_EQ(0.0, co->get());
+
+    // Single shot timer with minimum allowed interval of 20ms
+    EXPECT_TRUE(evaluateAndAssert(
+            R"(engine.beginTimer(20, function() {
+                   engine.setValue('[Test]', 'co', 1.0);
+               }, true);)"));
+    processEvents();
+    EXPECT_DOUBLE_EQ(0.0, co->get());
+
+    cEngine->thread()->msleep(35);
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(1.0, co->get());
+}
+
+TEST_F(ControllerScriptEngineLegacyTest, beginTimer_singleShotTimerBindFunction) {
+    auto co = std::make_unique<ControlPotmeter>(ConfigKey("[Test]", "co"),
+            -10.0,
+            10.0);
+    co->setParameter(0.0);
+    auto coTimerId = std::make_unique<ControlPotmeter>(ConfigKey("[Test]", "coTimerId"),
+            -10.0,
+            50.0);
+    coTimerId->setParameter(0.0);
+    EXPECT_TRUE(evaluateAndAssert(
+            "engine.setValue('[Test]', 'co', 0.0);"));
+    EXPECT_DOUBLE_EQ(0.0, co->get());
+
+    // Single shot timer with minimum allowed interval of 20ms
+    EXPECT_TRUE(evaluateAndAssert(
+            R"(var globVar = 7;
```

Comment:
lets use `let` everywhere for predictable scoping rules.

Reply:
I think `var` is required here - if I change this to `let`, the tests fail

Reply:
thats a little weird, what error?

Reply:
If I change only line 691 from `var `to `let`, I get:
```
 ControllerScriptEngineLegacyTest.beginTimer_singleShotTimerBindFunction
   Source: controllerscriptenginelegacy_test.cpp line 676
   Duration: 72 ms

  Message: 

warning [Main] "Script tried setting ([Test], co) to NotANumber (NaN)"
#1 - Expected equality of these values:
  7.0
    Which is: 7
  co->get()
    Which is: 0
warning [Main] "Script tried setting ([Test], co) to NotANumber (NaN)"
#2 - Expected equality of these values:
  8.0
    Which is: 8
  co->get()
    Which is: 0

  Stack Trace: 
#1 - controllerscriptenginelegacy_test.cpp:707 line 707
#2 - controllerscriptenginelegacy_test.cpp:711 line 711

```